src/Microsoft.Framework.Caching.SqlServer/SqlServerCache.cs

```diff
@@ -0,0 +1,478 @@
+                }
+            }
+
+            StartScanForExpiredCacheItems();
+
+            return cacheItemvalue;
+        }
+
+        public async Task<byte[]> GetAsync([NotNull] string key)
+        {
+            byte[] cacheItemvalue = null;
+            var utcNow = _systemClock.UtcNow;
+
+            using (var connection = new SqlConnection(_options.ConnectionString))
+            {
+                var command = new SqlCommand(_sqlQueries.GetCacheItem, connection);
+                command.Parameters.AddWithValue(CacheItemId, key);
+                command.Parameters.AddWithValue("UtcNow", utcNow);
+
+                await connection.OpenAsync();
+
+                var reader = await command.ExecuteReaderAsync(CommandBehavior.SingleRow);
+
+                DateTimeOffset oldExpirationTime;
+                TimeSpan? slidingExpiration = null;
+                DateTimeOffset? absoluteExpiration = null;
+                if (await reader.ReadAsync())
+                {
+                    oldExpirationTime = await reader.GetFieldValueAsync<DateTimeOffset>(_expiresAtTimeUTCIndex);
+                    cacheItemvalue = await reader.GetFieldValueAsync<byte[]>(_cacheItemValueIndex);
+
+                    if (!await reader.IsDBNullAsync(_slidingExpirationInTicksIndex))
+                    {
+                        slidingExpiration = TimeSpan.FromTicks(
+                            await reader.GetFieldValueAsync<long>(_slidingExpirationInTicksIndex));
+                    }
+
+                    if (!reader.IsDBNull(_absoluteExpirationIndex))
+                    {
+                        absoluteExpiration = await reader.GetFieldValueAsync<DateTimeOffset>(_absoluteExpirationIndex);
+                    }
+                }
+                else
+                {
+                    return null;
+                }
+
+                // Check if the current item has a sliding expiration and is going to expire within the
+                // next timeout period(ex:20 minutes), then extend the expiration time.
+                if (slidingExpiration.HasValue &&
+                    utcNow >= (oldExpirationTime - slidingExpiration.Value))
+                {
+                    var newExpirationTime = GetExpirationTime(slidingExpiration, absoluteExpiration, utcNow);
+
+                    if (oldExpirationTime != newExpirationTime)
+                    {
+                        reader.Dispose();
+
+                        command = new SqlCommand(_sqlQueries.UpdateCacheItemExpiration, connection);
+                        command.Parameters.AddWithValue(CacheItemId, key);
+                        command.Parameters.AddWithValue(ExpiresAtTimeUTC, newExpirationTime);
+
+                        await command.ExecuteNonQueryAsync();
+                    }
+                }
+            }
+
+            StartScanForExpiredCacheItems();
+
+            return cacheItemvalue;
+        }
+
+        public void Refresh([NotNull] string key)
```

Comment:
Consider giving `Refresh()` and `RefreshAsync()` its own implementation. It seems it shouldn't need to do half of what `Get()`/`GetAsync()` does.